Close #12165: Use Coords for Vehicle functions

heliobatimarqui opened the pull request:
Minor alterations in function prototypes and calls.

tupaschoal commented on contributors.md:
```diff
@@ -65,6 +65,7 @@ The following people are not part of the development team, but have been contrib
 * Kelson Blakewood (spacek531) - title sequences for release versions 0.0.4 through 0.1.2, title sequences using milliseconds
 * Hugo Wallenburg (Goddesen) - Misc.
 * Matte Andersson (Nubbie) - Misc, UX
+* Helio Batimarqui (heliobatimarqui) - Misc.
```
This list is kind of sorted by contribution date, so please move yours to the end

heliobatimarqui replied:
Done.

tupaschoal commented on src/openrct2/ride/Vehicle.cpp:
```diff
@@ -7243,7 +7243,7 @@ void Vehicle::UpdateAdditionalAnimation()
                             int32_t directionIndex = sprite_direction >> 1;
                             auto offset = SteamParticleOffsets[typeIndex][directionIndex];
-                            steam_particle_create(x + offset.x, y + offset.y, z + offset.z);
+                            steam_particle_create(CoordsXYZ{x + offset.x, y + offset.y, z + offset.z});
```
```suggestion
                            steam_particle_create(CoordsXYZ{x, y, z} + offset});
```

heliobatimarqui replied:
The type of offset is not CoordsXYZ nor a type that there is a + operator available.

tupaschoal commented on src/openrct2/ride/Vehicle.cpp:
```diff
@@ -7173,10 +7173,10 @@ void Vehicle::UpdateSpinningCar()
  *  rct2: 0x006734B2
  */
-static void steam_particle_create(int16_t x, int16_t y, int16_t z)
+static void steam_particle_create(const CoordsXYZ& coordinates)
```
```suggestion
static void steam_particle_create(const CoordsXYZ& coords)
```

We use coords often in the code, so I guess this shortened one would be nice.

heliobatimarqui replied:
Done as suggested!

tupaschoal commented on src/openrct2/ride/Vehicle.cpp:
```diff
@@ -7243,7 +7243,7 @@ void Vehicle::UpdateAdditionalAnimation()
                             int32_t directionIndex = sprite_direction >> 1;
                             auto offset = SteamParticleOffsets[typeIndex][directionIndex];
-                            steam_particle_create(CoordsXYZ{x + offset.x, y + offset.y, z + offset.z});
+                            steam_particle_create({x + offset.x, y + offset.y, z + offset.z});
```
This is kind of unrelated to this commit in specific.

tupaschoal commented:
Can you pleasereword the threefirst commits also? To something like:
`Part of #12165: Use CoordsXYZ on steam_particle_create`

heliobatimarqui commented:
> Can you rebase instead of merging, please? You can use it to squash commits together, drop others, etc.
> There is a tutorial here showing how to do it: https://github.com/OpenRCT2/OpenRCT2/wiki/Rebase-and-Sync-fork-with-OpenRCT2
> 
> I'm expecting 4 commits:
> 
> * Part of #12165: Use CoordsXYZ on steam_particle_create
> * Part of #12165: Use CoordsXYZ on try_add_synchronised_station function
> * Part of #12165: Use CoordsXYZ on vehicle_check_collision function
> * Add heliobatimarqui to contributors.md
> 
> Each commit should only contain modifications relative to that change (what we call atomic changesets)
> Please do mark issues as resolved when you deal with them too :)

Hi. I think I did it right now. And thank you for the tutorial.

heliobatimarqui commented:
> > Hi. I think I did it right now. And thank you for the tutorial.
> 
> You're welcome, we're almost there! Three comments:
> 
> 1. Can you please mark the comments that I opened as resolved, if you did so? This is a good practice where you show the reviewers that you cared for their suggestions and implemented (or not, but if you didn't, elaborating why is nice.
> 2. Are you facing any issues that are taking you to close and reopen the PR?
> 3. Can you split the changes done in the fifth commit to the commit where they actually belong? This means that the issue you solved on `CONTRIBUTING.md` should be on the 4th commit, the `try_add_synchronised_station` formatting fixes should be on the second commit and so on. You will need to do different commits for each of those fixes and squash them or ammend the commits. I usually do the former, so I typically would have this on my `git rebase -i develop` result:
> 
> ```
> pick d94e78 Part of #12165: Use CoordsXYZ on steam_particle_create
> pick 4e9baa Part of #12165: Use CoordsXYZ on try_add_synchronised_station function
> pick afb581 Part of #12165: Use CoordsXYZ on vehicle_check_collision function
> pick 643d0e Add heliobatimarqui to contributors.md
> pick 87871a Fix for contributors
> pick 0c3317 Fix for try_add_syncronised_station
> ```
> 
> Then I can reorder the commits and do squash/fixup on some of them, which will look like:
> 
> ```
> pick d94e78 Part of #12165: Use CoordsXYZ on steam_particle_create
> pick 4e9baa Part of #12165: Use CoordsXYZ on try_add_synchronised_station function
> fixup 0c3317 Fix for try_add_syncronised_station
> pick afb581 Part of #12165: Use CoordsXYZ on vehicle_check_collision function
> pick 643d0e Add heliobatimarqui to contributors.md
> fixup 87871a Fix for contributors
> ```
> 
> After saving, I'll end up with exactly this, where the fixup changes are incorporated on the commit above them
> 
> ```
> pick d94e78 Part of #12165: Use CoordsXYZ on steam_particle_create
> pick 4e9baa Part of #12165: Use CoordsXYZ on try_add_synchronised_station function
> pick afb581 Part of #12165: Use CoordsXYZ on vehicle_check_collision function
> pick 643d0e Add heliobatimarqui to contributors.md
> ...
> ```
Hi. It seems that I finally did it properly. 
1) Okay, I'll do this!
2) Yes, as it's the first time that I use git I'm facing some difficulty with it. But it seems that after your tutorial and some research I could do it, correct me if I'm wrong, please.
3) Fixed.

Again, thank you for the tutorial and thanks for the pacience too.

tupaschoal commented:
No problem, can you just reword the fourth commit to "Add heliobatimarqui to contributors.md" please?

heliobatimarqui commented:
> No problem, can you just reword the fourth commit to "Add heliobatimarqui to contributors.md" please?

Done!

tupaschoal commented:
One tip, always make changes on a branch of your fork and keep `develop` untouched. This will prevent you from having headaches in the future

heliobatimarqui commented:
> One tip, always make changes on a branch of your fork and keep `develop` untouched. This will prevent you from having headaches in the future

Thank you. I'll do it next time.

duncanspumpkin commented on src/openrct2/ride/Vehicle.cpp:
```diff
@@ -2690,15 +2690,15 @@ static rct_synchronised_vehicle* _lastSynchronisedVehicle = nullptr;
  *  rct2: 0x006DE1A4
  */
-static bool try_add_synchronised_station(int32_t x, int32_t y, int32_t z)
+static bool try_add_synchronised_station(const CoordsXYZ& coords)
```
This function is slightly wrong the coordinates of z is in small Z and not big Z. We need to change all callers and functions that this calls.

tupaschoal replied:
Fixed on #12192
